src/mme-app/actionHandlers/defaultMmeProcedureActionHandlers.cpp

```diff
@@ -1060,3 +1045,152 @@ ActStatus ActionHandlers::handle_detach_failure(ControlBlock& cb)
     return rc;
 }
 
+/***************************************
+* Action handler : default_delete_subs_req_handler
+***************************************/
+ActStatus ActionHandlers::default_delete_subs_req_handler(ControlBlock& cb)
+{
+    log_msg(LOG_DEBUG, "default_delete_subs_req_handler : Entry\n");
+
+    MsgBuffer* msgBuf = static_cast<MsgBuffer*>(cb.getMsgData());
+    VERIFY(msgBuf, return ActStatus::PROCEED, "Invalid message buffer\n");
+
+    const dsr_Q_msg_t* msgData_p = static_cast<const dsr_Q_msg_t *>(msgBuf->getDataPointer());
+
+    DigitRegister15 IMSI;
+    IMSI.setImsiDigits((unsigned char *)msgData_p->header.IMSI);
+
+    s6a_dsa_Q_msg dsa;
+    dsa.msg_type = delete_subscriber_data_answer;
+    dsa.eteId = msgData_p->eteId;
+    dsa.res_code = DIAMETER_UNABLE_TO_COMPLY;
+
+    UEContext *ueCtxt = static_cast<UEContext*>(cb.getPermDataBlock());
+    if (ueCtxt != NULL)
+    {
+        MmContext* mmCtxt = ueCtxt->getMmContext();
+        if(mmCtxt)
+        {
+            if (msgData_p->dsr_flags & dsrCompApnProfWdwl_c)
+            {
+                log_msg(LOG_INFO, "DSR received with Complete APN Config "
+                    "Profile Withdrawal Flag set to 1");
+            }
+            else
+            {
+                if (mmCtxt->getMmState() == EpsDetached)
+                {
+                    log_msg(LOG_INFO,
+                            "Subscriber is already detached. "
+                            "Cleaning up the contexts. UE IDx %d",
+                            cb.getCBIndex());
+                    dsa.res_code = DIAMETER_ERROR_USER_UNKNOWN;
+
+                    // cleanup the context??
+                    MmeContextManagerUtils::deleteUEContext(cb.getCBIndex());
+                }
+                else
+                {
+                    if(msgData_p->dsr_flags & dsrPdnSubsCtxWdwl_c)
+                    {
+                        // check if apn context id matches with any in the session,
+                        // if yes, trigger Detach.
+                        SessionContext* sess_p = ueCtxt->findSessionContextByApnConfigProfileCtxId(msgData_p->apn_conf_prof_id);
+                        if(sess_p)
+                        {
+                            dsa.res_code = DIAMETER_SUCCESS;
+
+                            MmeDetachProcedureCtxt* prcdCtxt_p =
+                                    MmeContextManagerUtils::allocateDetachProcedureCtxt(cb, hssInitDetach_c);
+                            if(prcdCtxt_p != NULL)
+                            {
+                                prcdCtxt_p->setNasDetachType(reattachNotRequired);
```

Comment:
@thakurajayL @badhrinathpa 
Hi Ajay, Badhri,
Could you please clarify , if we can set the detachType as "reattachNotRequired" here?

Reply:
In case HSS is deleting profile, we should not set re-attach requested.  So whatever you have set seems to be correct.

Reply:
Okay, thank you.